pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -302,6 +225,25 @@ func newKubeVirtConfigForCR(cr *hcov1alpha1.HyperConverged) *corev1.ConfigMap {
 	}
 }
 
+func createKubeVirtConfigCRAndSetOwnership(instance *hcov1alpha1.HyperConverged, request reconcile.Request, r *ReconcileHyperConverged, c chan CreateCRResult) {
+	kvConfig := newKubeVirtConfigForCR(instance)
```

Comment:
is it possible to simply have one function, like `createCrAndSetOwnership` that (in addition to what I see here) takes the name (ie. `KubeVirtConfig`) and the CR generator function (ie. `newKubeVirtConfigForCR`)?

Reply:
Let me look into this again. I had tried before, but didn't find a compatible type. Looking at the add function simplification you submitted, I wonder if we can also use runtime.Object here.